### PR TITLE
Callback for Unity to receive session token for continuity

### PR DESCRIPTION
**Pull request by Div-Inworld**

### CODE REVIEW

File: src/Client.h [hunk 1]
```diff
@@ -81,6 +81,10 @@ namespace Inworld
 		ConnectionState GetConnectionState() const { return _ConnectionState; }
 		bool GetConnectionError(std::string& OutErrorMessage, int32_t& OutErrorCode) const;
 
+#ifdef INWORLD_UNITY
+		std::function<void(SessionInfo)> UnitySessionTokenCallback;
```
Comment (art-vozniuk) [inline src/Client.h:85]: after token is generated, it's cached in SessionInfo field. it seems like you just need a getter for it. let's discuss

**Review by art-vozniuk — CHANGES_REQUESTED** [review]

File: src/Client.h (hunk 1 above)
Comment (Div-Inworld) [reply src/Client.h]: Implemented refactor, added getter leveraging GenerateToken callback and removing ifdefs

File: Example/Unity/client_wrapper.cpp [hunk 2]
```diff
@@ -97,61 +130,47 @@ extern "C" {
...
+        {
+            Inworld::Log("Session info is valid in NDK, starting client");
+            wrapper->client.StartClient(opt, session_info, agentInfosHandler);
+        }
+        else
+        {
+            Inworld::Log("Session info is invalid in NDK, starting client");
+            wrapper->client.GenerateToken([SessionTokenCallback, wrapper, opt, agentInfosHandler]()
+            {
+                auto info = wrapper->client.GetSessionInfo();
```
Comment (art-vozniuk) [inline Example/Unity/client_wrapper.cpp:158]: const auto& info = .... to avoid copy

File: src/Client.h [hunk 3]
```diff
@@ -51,6 +51,8 @@ namespace Inworld
 		ClientBase() = default;
 		virtual ~ClientBase() = default;
 
+		SessionInfo GetSessionInfo();
```
Comment (art-vozniuk) [inline src/Client.h:54]: const SessionInfo& GetSessionInfo() const

File: src/Client.h [hunk 4]
```diff
@@ -51,6 +51,8 @@ namespace Inworld
...
+		void SetOptions(ClientOptions options);
```
Comment (art-vozniuk) [inline src/Client.h:55]: void SetOptions(const ClientOptions& Options);

and move both of methods lower, to the end of public section please

File: Example/Unity/client_wrapper.cpp [hunk 5]
```diff
@@ -50,15 +50,48 @@ extern "C" {
         DebugLog("INITIALIZED CLIENT FROM DLL");
     }
 
+    void HandleAgentInfos(const std::vector<Inworld::AgentInfo>& AgentInfos, LoadSceneCallbackType LoadSceneCallback) {
```
Comment (art-vozniuk) [inline Example/Unity/client_wrapper.cpp:53]: if it's used only in this cpp file then
static void Handle...

**Review by art-vozniuk — CHANGES_REQUESTED** [review]

File: src/Client.h (hunk 4 above)
Comment (art-vozniuk) [reply src/Client.h]: you didn't fix void SetOptions(const ClientOptions& Options);

**Review by art-vozniuk — APPROVED** [review]

**Pull request merged**